p2p/sql_stmt.go

```diff
@@ -48,24 +70,167 @@ func (srv *Server) initSql() error {
 		if err := srv.prepareGetRowID(); err != nil {
 			return err
 		}
-	} else {
-		log.Trace("No sql db connection info provided")
 	}
 	return nil
 }
 
+func (srv *Server) checkIfTableExists(dbName string, tableName string) (int, error) {
+	var result int
+	err := srv.DB.QueryRow(`
+		SELECT COUNT(*) 
+		FROM information_schema.TABLES 
+		WHERE (TABLE_SCHEMA = ?) AND (TABLE_NAME = ?)
+	`, dbName, tableName).Scan(&result)
+	return result, err
+}
+
+func (srv *Server) backupTable(dbName string, tableName string) error {
+	// check if table exists
+	if result, err := srv.checkIfTableExists(dbName, tableName); err != nil {
+		log.Error(fmt.Sprintf("Failed to check if %s table exists", tableName), "database", srv.MySQLName, "err", err)
+		return err
+	} else if result == 0 {
+		log.Debug(tableName+" table not found. Skipping backup", "database", srv.MySQLName)
+		return nil
+	}
+	log.Trace(tableName+" table exists", "database", srv.MySQLName)
+
+	// flush table
+	if _, err := srv.DB.Exec(fmt.Sprintf(`
+		FLUSH TABLES %s WITH READ LOCK
```

Comment:
What scenario is the read lock protecting from here?

Reply:
tl;dr I will take out FTWRL (flush tables with read lock) and `UNLOCK TABLES` parts.

I originally thought using FTWRL (flush tables with read lock) before making backups was the best practice. It turns out there really is no point of using FTWRL for InnoDB (https://dba.stackexchange.com/questions/52147/can-i-time-out-my-scripted-flush-tables-with-read-lock). It could actually result in deadlock/long-wait-time if there happens to be any complicated query (https://www.percona.com/blog/2012/03/23/how-flush-tables-with-read-lock-works-with-innodb-tables/)

The scenario I was really considering was: `node-finder` crashes for whatever reason, tries to make a backup of the tables upon restart, and there are some pending `select` statements from `eth-monitor`. I thought this would make sure `node-finder` waits for the queries to complete. This is wrong; I would want to use FTWRL and UNLOCK around `dropTables` if I wanted to make sure `eth-monitor` receive correct results.

Anyways, I'm removing lock/unlocking tables entirely. I think it makes more sense for `eth-monitor` to handle query errors itself, rather than trying to avoid the situation from `node-finder` side.